animate/utility.py

```diff
@@ -154,19 +153,20 @@ def norm(v, norm_type="L2", condition=None, boundary=False):
             if p < 1:
                 raise ValueError(f"'{norm_type}' norm does not make sense.")
             integrand = ufl.inner(v, v)
-        elif norm_type.lower() == "h1":
-            integrand = ufl.inner(v, v) + ufl.inner(ufl.grad(v), ufl.grad(v))
-        elif norm_type.lower() == "hdiv":
-            integrand = ufl.inner(v, v) + ufl.div(v) * ufl.div(v)
-        elif norm_type.lower() == "hcurl":
-            integrand = ufl.inner(v, v) + ufl.inner(ufl.curl(v), ufl.curl(v))
+        elif norm_type.lower() in ("h1", "hdiv", "hcurl"):
+            integrands_dict = {
+                "h1": ufl.inner(v, v) + ufl.inner(ufl.grad(v), ufl.grad(v)),
+                "hdiv": ufl.inner(v, v) + ufl.div(v) * ufl.div(v),
+                "hcurl": ufl.inner(v, v) + ufl.inner(ufl.curl(v), ufl.curl(v)),
+            }
+            integrand = integrands_dict[norm_type.lower()]
```

Comment:
Again here. Also, these could be functions rather than passing `v` to them all.

Reply:
Done as you suggested in 2946fc8a3241de5d1fa4732cda88d6a1ad16c0cf